### DISCUSSION TITLE
Dyskusja o tutorialach API (helloroman-api-course)

### DISCUSSION THREAD

**Discussion by helloroman** [helloroman/helloroman-api-course#1] [General]
## 👋  Siema!
  Jeśli macie jakiekolwiek uwagi, pomysły, zastrzeżenia dotyczące kursu, to ta zakładka jest dla Was! Jeśli macie jakiś problem z kursem, lub podejściem, które prezentuję, to będę zachwycony jeśli będziemy mogli o tym tutaj podyskutować ♥️

**Comment by olgasowlet** [comment]
Hej! Jestem tu nowa i widzę, że nikt nie używa opcji dyskusji na GitHubie a skoro chciałeś to przetestować to oto jestem! hehe 

Chciałam dopytać o DocumentFragment. 

Jak przerabiałam pierwszy odcinek o API to starałam się pisać sama kod a potem porównywać z twoim. Nie znałam DocumentFragment i stworzyłam coś takiego:

```
const beersElement = document.querySelector(".container");

const renderBeers = (beers) => {
    let htmlElement = ' ';

    beers.forEach(beer => {
        htmlElement += `
            <div class="beer">
                <div class="beer--content">
                    <h1 class="beer--title">${beer.name}</h1>
                    <p class="beer--tagline">${beer.tagline}</p>
                    <p class="beer--description">${beer.description}</p>
                </div>
                <img class="beer--image" src="${beer.image_url}">
            </div>
        `;
    });

    beersElement.innerHTML = htmlElement;
};
```

W filmiku argumentujesz używanie DocumentFragment tym, że DOM renderuje nam się tylko raz. W moim rozwiązaniu również tak jest. Czy jest ono okej, czy jednak i tak lepiej użyc DocumentFragment?

Z góry dzięki!

**Reply by helloroman** [reply]
Hej, super że piszesz! Nie jestem ekspertem od performance jeśli chodzi o operacje na DOM, ale znalazłem taki ciekawy wątek – https://www.reddit.com/r/javascript/comments/9d4i8d/when_to_use_innerhtml_and_is_it_slower_than/ 

Myślę, że przy tak prostym przykładzie różnice nie byłyby ogromne, ale w bardziej złożonych operacjach appendChild mogłoby okazać się szybsze
